doc/source/f2py/usage.rst

```diff
@@ -327,3 +327,68 @@ resulting package to work, you need to create a file named ``__init__.py``
 (in the same directory as ``add.pyf``). Notice the extension module is
 defined entirely in terms of the ``add.pyf`` and ``add.f`` files. The
 conversion of the .pyf file to a .c file is handled by `numpy.distutils`.
+
+Building with Meson (Examples)
+==============================
+
+Using f2py with Meson
+~~~~~~~~~~~~~~~~~~~~~
+
+Meson is a modern build system recommended for building Python extension
+modules, especially starting with Python 3.12 and NumPy 2.x. Meson provides
+a robust and maintainable way to build Fortran extensions with f2py.
+
+To build a Fortran extension using f2py and Meson, you can use Meson's
+`custom_target` to invoke f2py and generate the extension module. The
+following minimal example demonstrates how to do this:
+
+.. code-block:: meson
+
+   # List your Fortran source files
+   fortran_sources = files('your_module.f90')
```

Comment:
Rather than make this a generic "your_module", I would suggest to continue the example above, i.e., show how one can use `meson` to build the `f2py_examples` package.